src/replica.rs

```diff
@@ -391,11 +414,27 @@ impl Replica {
 
     /// Verify that this is a valid _propagated_
     /// DebitAgreementProof, i.e. signed by a group that we know of.
-    fn verify_propagated_proof(&self, proof: &DebitAgreementProof) -> Result<safe_nd::PublicKey> {
+    fn verify_propagated_proof<F: FnOnce() -> Option<PublicKey>>(
+        &self,
+        proof: &DebitAgreementProof,
+        f: F,
```

Comment:
Same as above (`f` is a very obscure naming).

Reply:
Yeh, totally agree here.